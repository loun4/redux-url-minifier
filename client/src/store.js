
import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { createForms } from 'react-redux-form';
import entities from './reducers/entities';

const loggerMiddleware = createLogger({
  predicate() { return process.env.NODE_ENV === 'development'; },
});

const forms = createForms({
  linkForm: {
    linkURL: '',
  },
  signinForm: {
    login: '',
    password: '',
  },
});

export default function configureStore(preloadedState) {
  return createStore(
    combineReducers({
      ...forms,
      entities,
    }),
    preloadedState,
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware,
    ),
  );
}
