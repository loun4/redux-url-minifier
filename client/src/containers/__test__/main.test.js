import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../store';
import Main from '../main';
import RouterProvider from '../router-provider';

test('Render main', () => {
  const subject = (
    <RouterProvider>
      <Provider store={store()}>
        <Main />
      </Provider>
    </RouterProvider>
  );

  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
