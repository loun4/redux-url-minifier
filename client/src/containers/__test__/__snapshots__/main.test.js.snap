// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render main 1`] = `
<div>
  <form
    className="ui form column massive separator"
    noValidate={undefined}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        className="ui action input"
      >
        <input
          autoComplete="off"
          className="field-input"
          disabled={false}
          id="link"
          name="linkForm.linkURL"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholder="Past your original url"
          type="text"
          value=""
        />
        <button
          className="ui button"
          type="submit"
        >
          Go !
        </button>
      </div>
    </div>
  </form>
</div>
`;
